managedservices/managedservices.go

```diff
@@ -0,0 +1,230 @@
+			return microerror.Mask(err)
+		}
+
+		ms.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("installed chart %#q", chartConfig.ChartName))
+	}
+
+	{
+		ms.logger.LogCtx(ctx, "level", "debug", "message", "waiting for deployed status")
+
+		err = ms.resource.WaitForStatus(chartConfig.ChartName, "DEPLOYED")
+		if err != nil {
+			return microerror.Mask(err)
+		}
+
+		ms.logger.LogCtx(ctx, "level", "debug", "message", "chart is deployed")
+	}
+	{
+		ms.logger.LogCtx(ctx, "level", "debug", "message", "checking resources")
+
+		for _, ds := range chartResources.DaemonSets {
+			ms.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("checking daemonset %#q", ds.Name))
+
+			err = ms.checkDaemonSet(ds)
+			if err != nil {
+				return microerror.Mask(err)
+			}
+
+			ms.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("daemonset %#q is correct", ds.Name))
+		}
+
+		for _, d := range chartResources.Deployments {
+			ms.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("checking deployment %#q", d.Name))
+
+			err = ms.checkDeployment(d)
+			if err != nil {
+				return microerror.Mask(err)
+			}
+
+			ms.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("deployment %#q is correct", d.Name))
+		}
+
+		ms.logger.LogCtx(ctx, "level", "debug", "message", "resources are correct")
+	}
+
+	{
+		ms.logger.LogCtx(ctx, "level", "debug", "message", "running release tests")
+
+		err = ms.helmClient.RunReleaseTest(chartConfig.ChartName)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+
+		ms.logger.LogCtx(ctx, "level", "debug", "message", "release tests passed")
+	}
+
+	return nil
+}
+
+// checkDaemonSet ensures that key properties of the daemonset are correct.
+func (ms *ManagedServices) checkDaemonSet(expectedDaemonSet DaemonSet) error {
+	ds, err := ms.hostFramework.K8sClient().Apps().DaemonSets(expectedDaemonSet.Namespace).Get(expectedDaemonSet.Name, metav1.GetOptions{})
+	if apierrors.IsNotFound(err) {
+		return microerror.Maskf(notFoundError, "daemonset: %#q", expectedDaemonSet.Name, err)
+	} else if err != nil {
+		return microerror.Mask(err)
+	}
+
+	if !reflect.DeepEqual(expectedDaemonSet.Labels, ds.ObjectMeta.Labels) {
+		return microerror.Maskf(invalidLabelsError, "expected labels: %v got: %v", expectedDaemonSet.Labels, ds.ObjectMeta.Labels)
+	}
+
+	if !reflect.DeepEqual(expectedDaemonSet.MatchLabels, ds.Spec.Selector.MatchLabels) {
+		return microerror.Maskf(invalidLabelsError, "expected match labels: %v got: %v", expectedDaemonSet.MatchLabels, ds.Spec.Selector.MatchLabels)
+	}
+
+	if !reflect.DeepEqual(expectedDaemonSet.Labels, ds.Spec.Template.ObjectMeta.Labels) {
+		return microerror.Maskf(invalidLabelsError, "expected pod labels: %v got: %v", expectedDaemonSet.Labels, ds.Spec.Template.ObjectMeta.Labels)
+	}
+
+	return nil
+}
+
+// checkDeployment ensures that key properties of the deployment are correct.
+func (ms *ManagedServices) checkDeployment(expectedDeployment Deployment) error {
```

Comment:
Is it possible to extend/reuse https://github.com/giantswarm/e2e-harness/blob/master/pkg/framework/deployment/deployment.go#L40 instead doing it here? (Same for daemonset)

Reply:
Yes I know you added the deployment package but I decided not to go with it. To me it feels a bit too low level and I'd need to also add a daemonset package.

If we can add a generic package to e2e-harness that can test any k8s resource that feels better to me. Possibly this can be done by using the K8s REST client. But I don't have time to do that for this change.